pkg/resources/nifi/servicemonitor.go

```diff
@@ -0,0 +1,68 @@
+package nifi
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/konpyutaika/nifikop/api/v1alpha1"
+	nifiutil "github.com/konpyutaika/nifikop/pkg/util/nifi"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+	"github.com/konpyutaika/nifikop/pkg/resources/templates"
+	"github.com/konpyutaika/nifikop/pkg/util"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Reconciler) serviceMonitor(log logr.Logger) (runtimeClient.Object, error) {
+	// ensure there is a prometheus port configured or else fail.
+	var found bool
+	var prometheusListener v1alpha1.InternalListenerConfig
+	for _, listener := range r.NifiCluster.Spec.ListenersConfig.InternalListeners {
```

Comment:
On second thought with this, should there be an optional `InternalListenerConfig` provided with the `serviceMonitor` config or should it continue to depend on there being a prometheus `spec.ListenersConfig.InternalListenerConfig` as its written?

What's the better user experience?